packages/compat/src/v1-appboot.ts

```diff
@@ -22,17 +22,19 @@ export class WriteV1AppBoot extends Plugin {
 
 export class ReadV1AppBoot extends Plugin {
   private appBoot: string | undefined;
+  private hasBuilt = false;
   constructor(appBootTree: Node) {
     super([appBootTree], {
       persistentOutput: true,
     });
   }
   build() {
     this.appBoot = readFileSync(join(this.inputPaths[0], `config/app-boot.js`), 'utf8');
+    this.hasBuilt = true;
   }
 
   readAppBoot() {
-    if (!this.appBoot) {
```

Comment:
FWIW, the prior guard had issues because in cases where no addon emits `{{content-for 'app-boot'}}` content the value will be `''`, which is going to evaluate to falsey.